---
title: "Hierarchical Gaussian process regression with application to microarray data"
author: "Spencer Woody"
date: "July 1, 2017"
output: 
    html_document:
        theme: cosmo
        highlight: textmate
---

```{r setup, include=FALSE}
library(ggplot2)
knitr::opts_chunk$set(echo = TRUE, fig.path = "img/")
```

## Introduction

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>. Here's some math: $$ \sum_{i=1}^N X_i $$ And now some inline math: $\epsilon_i  \sim_{\text{iid}} \mathcal{N}(0, 1)$.
 
When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

$$ 	\begin{aligned} h_i(t) &\sim \text{GP}(\mathbf{0}, k_h(t, t')) \\
	g_n(t) &\sim \text{GP}(h_i(t), k_g(t, t')) \text{ for } n\in c_i \\
	f_{nr}(t) &\sim \text{GP}(g_n(t), k_f(t, t')) \end{aligned} $$


```{r start}
# Resave the data with new format
drosophila <- read.csv("droslong.csv", header = T)
head(drosophila)

# Simpler dataframe
fruitfly <- data.frame(
	group = drosophila$group, 
	gene = drosophila$gene,
	replicate = drosophila$replicate, 
	time = drosophila$time,
	log2exp = drosophila$log2exp)
	
# Order the dataframe
fruitfly <- fruitfly[with(fruitfly, order(group, gene, replicate, time)), ]
head(fruitfly)
```

## Including Plots

You can also embed plots, for example:

```{r plot-all-genes}
allgenes <- ggplot(drosophila, aes(time, log2exp)) +
    geom_point(aes(col = replicate)) +
    facet_wrap(~gene) +
    ggtitle("Expression Profiles for all Genes") +
    xlab("Time") +
    ylab("log2-Normalized gene expression") + 
    scale_colour_brewer(palette="Set1") +
    scale_x_continuous(breaks = seq(0, 12, by = 2)) +
    theme_bw() +
    theme(plot.title = element_text(hjust = 0.5))
allgenes
```

```{r, echo = F, eval = F}
# Run this at the end to make a copy of the html output as index.html
system("cp HGP.html index.html")
```


Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
