---
title: "Hierarchical Gaussian process regression with application to microarray data"
author: "Spencer Woody"
date: "August 26, 2017"
output: 
    html_document:
        theme: readable
        highlight: textmate
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.path = "img/")
```

## Introduction

Based on a paper by [Hensman, Lawrence, and Rattray](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-252), we model gene expression levels as a hierarchical Gaussian process,

$$ 	\begin{aligned} h_i(t) &\sim \text{GP}(\mathbf{0}, k_h(t, t')) \\
	g_n(t) &\sim \text{GP}(h_i(t), k_g(t, t')) \text{ for } n\in c_i \\
	f_{nr}(t) &\sim \text{GP}(g_n(t), k_f(t, t')), \end{aligned}  $$
	
for some covariance functions $k_h(t, t')$, $k_g(t, t')$, and $k_f(t, t')$. 

Our data consist of a small subset of from a time-course DNA microarray experiment with *Drosophila melanogaster*.

## The math

Suppose we have $\mathbf{h}_i$, $\mathbf{g}_n$, and $\mathbf{f}_{nr}$, which are vectors of draws from draws from $h_i(t)$, $g_n(t)$, and $f_{nr}(t)$, respectively, at times $\mathbf{t}_{nr}$.

## Implementation in R

First we will library in the required packages and read in the data in the format we want, which reflects the hierarchical nature of the data. 

```{r start, warning=FALSE, message=FALSE}
library(tidyverse)
library(RColorBrewer)
theme_set(theme_bw(base_size = 14)) # Slightly larger text for ggplot2

# Read in data, rearrange columns & rows, change levels order in gene factor
drosophila <- read_csv("droslong.csv") %>%
    select(group, gene, replicate, time, log2exp) %>% 
    arrange(group, gene, replicate, time) %>% 
    mutate(gene = fct_inorder(gene))

head(drosophila)
```

### Plotting the data

You can also embed plots, for example:

```{r plot-all-genes, fig.width=7, fig.height=6}
allgenes <- drosophila %>% 
    ggplot(aes(time, log2exp)) +
    geom_point(aes(col = group, alpha = replicate)) +
    facet_wrap(~gene) +
    labs(title = "Expression Profiles for all Genes",
         x = "Time",
         y = "log2-Normalized gene expression") + 
    scale_colour_brewer(palette = "Dark2") +
    scale_x_continuous(breaks = seq(0, 12, by = 2)) 
allgenes

group1plot <- drosophila %>% 
    filter(group == "group1") %>% 
    ggplot(aes(time, log2exp)) +
    geom_point(aes(col = gene)) +
    facet_wrap(~gene) +
    labs(title = "Expression Profiles for all Genes",
         x = "Time",
         y = "log2-Normalized gene expression") + 
    scale_colour_brewer(palette = "Set1") +
    scale_x_continuous(breaks = seq(0, 12, by = 2)) +
    theme(plot.title = element_text(hjust = 0.5)) +
    scale_y_continuous(limits = c(min(drosophila$log2exp),
                                  max(drosophila$log2exp)))

group2plot <- drosophila %>% 
    filter(group == "group2") %>% 
    ggplot(aes(time, log2exp)) +
    geom_point(aes(col = gene)) +
    facet_wrap(~gene) +
    labs(title = "Expression Profiles for all Genes",
         x = "Time",
         y = "log2-Normalized gene expression") + 
    scale_colour_brewer(palette = "Set1") +
    scale_x_continuous(breaks = seq(0, 12, by = 2)) +
    scale_y_continuous(limits = c(min(drosophila$log2exp),
                                  max(drosophila$log2exp)))

```


Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

## Appendix: R session information

```{r}
sessionInfo()
```

```{r, echo = F, eval = F}
# Run this at the end to make a copy of the html output as index.html
system("cp HGP.html index.html")
```


